addons/mail/models/mail_thread.py

```diff
@@ -1805,25 +1805,33 @@ def _message_add_suggested_recipient(self, result, partner=None, email=None, lan
                 reason, create_values """
         self.ensure_one()
         partner_info = {}
+        recipient_data = {'lang': lang, 'reason': reason}
         if email and not partner:
             # get partner info from email
             partner_info = self._message_partner_info_from_emails([email])[0]
             if partner_info.get('partner_id'):
                 partner = self.env['res.partner'].sudo().browse([partner_info['partner_id']])[0]
-        if email and email in [val[1] for val in result[self.ids[0]]]:  # already existing email -> skip
+        if email and email in [val['email'] for val in result[self.ids[0]]]:  # already existing email -> skip
             return result
         if partner and partner in self.message_partner_ids:  # recipient already in the followers -> skip
             return result
-        if partner and partner.id in [val[0] for val in result[self.ids[0]]]:  # already existing partner ID -> skip
+        if partner and partner.id in [val['persona'].get('id') for val in result[self.ids[0]] if
+                                      val.get('persona', {}).get('type') == 'partner']:  # already existing partner ID -> skip
             return result
         if partner and partner.email:  # complete profile: id, name <email>
-            result[self.ids[0]].append((partner.id, partner.email_formatted, lang, reason, {}))
+            email_normalized = ','.join(email_normalize_all(partner.email))
+            recipient_data.update(
+                {'name': partner.name, 'email': email_normalized, 'persona': {'type': 'partner', 'id': partner.id}})
```

Comment:
Please don't introduce useless layers and stay coherent with the rest of Odoo. Persona means nothing outside of discuss "ORM", just stay with classic keys easier to use: partner_id. It can be either False, either an id, indicating we have one, or not.
```suggestion
                {'name': partner.name, 'email': email_normalized, 'partner_id': partner.id})
```